src/main/java/com/speaker/repository/MessageRepositoryImpl.java

```diff
@@ -26,10 +28,12 @@ public List<Message> findAllByAccountId(int id) {
     }
 
     @Override
-    public int insert(Message message) {
-        return Objects.requireNonNull(dsl.insertInto(MESSAGE, MESSAGE.FROM_ACCOUNT_ID, MESSAGE.TO_ACCOUNT_ID, MESSAGE.DATE, MESSAGE.TEXT, MESSAGE.STATUS)
-                .values(message.getFromAccountId(), message.getToAccountId(), Timestamp.valueOf(LocalDateTime.now()), message.getText(), message.getStatus().toString())
-                .returningResult(MESSAGE.ID).fetchOne()).get(MESSAGE.ID);
+    public Optional<List<Integer>> insert(List<Message> message) {
+        return Optional.of(message.stream()
+                .map(ms -> Objects.requireNonNull(dsl.insertInto(MESSAGE, MESSAGE.FROM_ACCOUNT_ID, MESSAGE.TO_ACCOUNT_ID,
+                                MESSAGE.DATE, MESSAGE.TEXT, MESSAGE.STATUS)
+                        .values(ms.getFromAccountId(), ms.getToAccountId(),
+                                Timestamp.valueOf(LocalDateTime.now()), ms.getText(), ms.getStatus().toString())
+                        .returningResult(MESSAGE.ID).fetchOne()).get(MESSAGE.ID)).collect(Collectors.toList()));
```

Comment:
one execute per row in DB - is a bad way, if we want to create many rows in one operation need to use [batch operations](https://www.jooq.org/doc/latest/manual/sql-execution/batch-execution/)

Reply:
comments accepted, corrections made. Added a list of messages in one query to the database

Reply:
Comments accepted, corrections made. Add batch function and list of query